torch/utils/ordered_set.py

```diff
@@ -0,0 +1,174 @@
+from __future__ import annotations
+
+from collections.abc import MutableSet, Set as AbstractSet
+from typing import (
+    Any,
+    cast,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+)
+
+T = TypeVar("T")
+T_co = TypeVar("T_co", covariant=True)
+
+__all__ = ["OrderedSet"]
+
+
+class OrderedSet(MutableSet[T]):
+    """
+    Insertion ordered set, similar to OrderedDict.
+    """
+
+    __slots__ = ("_dict",)
+
+    def __init__(self, iterable: Optional[Iterable[T]] = None):
+        self._dict: dict[T, None] = {}
+        if iterable is not None:
+            self._dict = dict.fromkeys(iterable, None)
+
+    @staticmethod
+    def from_dict(dict_inp: Dict[T, None]) -> OrderedSet[T]:
```

Comment:
```suggestion
    def _from_dict(dict_inp: Dict[T, None]) -> OrderedSet[T]:
```

It should be an implementation detail and be private. The public "from_dict" semantic should be:

```python
OrderdedSet(dict_)
```